### PR TITLE
GenerateTag function to separate GenerateFullyQualifiedImageName

### PR DESCRIPTION
**Pull request by felixtran39**
**Description**
This is a non-functional change. Previously, taggers only had the option to generate the entire image name. This change provides a method for taggers to only generate the tag portion. This refactoring is in preparation for another change.

**Follow-up Work**
This refactoring is in preparation for the feature associated with issue #4371

### CODE REVIEW

File: pkg/skaffold/build/tag/custom.go [hunk 1]
```diff
@@ -27,18 +27,28 @@ type CustomTag struct {
...
+func (t *CustomTag) GenerateTag(workingDir, imageName string) (string, error) {
+	tag := t.Tag
 	if tag == "" {
 		return "", errors.New("custom tag not provided")
 	}
+	return tag, nil
+}
+
+// GenerateFullyQualifiedImageName tags an image with the custom tag.
+func (t *CustomTag) GenerateFullyQualifiedImageName(workingDir, imageName string) (string, error) {
```
Comment (briandealwis) [inline pkg/skaffold/build/tag/custom.go:47]: Maybe worth creating a common function for most of the implementations to call?  (except the envTemplateTagger)

File: pkg/skaffold/build/tag/env_template.go [hunk 2]
```diff
@@ -69,3 +69,12 @@ func (t *envTemplateTagger) GenerateFullyQualifiedImageName(workingDir, imageNam
...
 	return tag, nil
 }
+
+// GenerateFullyQualifiedImageName tags an image with the custom tag
+func (t *envTemplateTagger) GenerateFullyQualifiedImageName(workingDir, imageName string) (string, error) {
+	tag, err := t.GenerateTag(workingDir, imageName)
+	if err != nil {
+		return "", fmt.Errorf("generating tag: %w", err)
+	}
+	return tag, nil
```
[the remark this answers is not in the document]
Comment (felixtran39) [reply pkg/skaffold/build/tag/env_template.go:79]: This change would not work with how envTemplate is currently designed. Is this a change that you still want?

File: pkg/skaffold/build/tag/custom.go (hunk 1 above)
Comment (felixtran39) [reply pkg/skaffold/build/tag/custom.go:47]: I will be creating a common function for all implementations to call. We will be deprecating envTemplateTagger's use of {{.IMAGE_NAME}} and assume the template only specifies the tag value.

**Comment by felixtran39** [comment]
Closing PR for now

**Pull request closed without merging**